app/src/main/java/com/example/allclear/auth/UsaintLoadingActivity.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    //로그인 후 토큰을 저장하는 함수
+    private void tokenSave(String accessToken,String refreshToken, Long memberId){
+        preferences = getSharedPreferences(DB, MODE_PRIVATE);
+        Log.i("access",accessToken);
+        Log.i("re",refreshToken);
+        Log.i("memId", ""+memberId);
+        MemberId = memberId;
+        AccessToken = accessToken;
+        preferences.edit().putString(ACCESS_TOKEN,accessToken).apply();
+        preferences.edit().putString(REFRESH_TOKEN,refreshToken).apply();
+        preferences.edit().putLong(USER_ID,memberId).apply();
+
+        updateUser();
+    }
+
+    //학생 정보를 업데이트 하는 함수
+    void updateUser(){
+        binding.icStudent.setImageTintList(ContextCompat.getColorStateList(getApplicationContext(), R.color.first_blue)); //아이콘 색상 변경
+        binding.tvStudent.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.first_blue)); //텍스트 색상 변경
+
+        usaintUpdateRequestDto = new UsaintUpdateRequestDto(usaintId, usaintPassword);
+        Call<TestResponseDto> call = updateUserService.updateUser("Bearer " + AccessToken , MemberId, usaintUpdateRequestDto);
+        serverCommunication(call, this::updateScore);
+    }
+
+    //성적을 업데이트하는 함수
+    void updateScore(){
+        binding.icGrade.setImageTintList(ContextCompat.getColorStateList(getApplicationContext(), R.color.first_blue)); //아이콘 색상 변경
+        binding.tvGrade.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.first_blue)); //텍스트 색상 변경
+
+        Call<TestResponseDto> call = gradeAndCurriculumUpdateService.updateGradeAndCurriculum("Bearer " + AccessToken , MemberId, usaintUpdateRequestDto);
+        serverCommunication(call, this::updateRequirement);
+    }
+
+    //졸업 요건을 업데이트하는 함수
+    void updateRequirement(){
+        binding.icGraduation.setImageTintList(ContextCompat.getColorStateList(getApplicationContext(), R.color.first_blue)); //아이콘 색상 변경
+        binding.tvGraduation.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.first_blue)); //텍스트 색상 변경
+
+        Call<TestResponseDto> call = updateRequirementService.updateRequirement("Bearer " + AccessToken , MemberId, usaintUpdateRequestDto);
+        serverCommunication(call, this::AfterSycn);
+    }
+
+    //동기화 완료 함수
+    void AfterSycn(){
+        progressBar.setVisibility(View.GONE); //로딩바 숨기기
+        Intent intent = new Intent(getApplicationContext(), MainPageActivity.class);
+        startActivity(intent);
+    }
+
+    // 서버와 통신하는 함수
+    void serverCommunication(Call<TestResponseDto> call, Runnable onSuccess) {
+        call.enqueue(new Callback<TestResponseDto>() {
+            @Override
+            public void onResponse(Call<TestResponseDto> call, Response<TestResponseDto> response) {
+                if (response.isSuccessful()) {
+                    onSuccess.run();
+                } else {
+                    if (response.body() != null) {
+                        signUpResultHandler(response.body().getCode()); //동기화 실패시 예외 처리
```

Comment:
```suggestion
                    if (response.body() != null) {
                        Toast.makeText(getApplicationContext(), response.body().getMessage().toString(), Toast.LENGTH_SHORT).show();
```

음,, 될 지는 모르겠지만,,제 생각에는 예외처리를 각 코드에 대한 토스트를 띄우는 것보다 
서버에서 보내준 메세지를 바로 띄우는 방법도 가능할 것 같아서 한번 제안해봅니다!!

Reply:
코드가 줄어서 좋을거 같아요!